src/adapter/src/coord.rs

```diff
@@ -755,6 +765,7 @@ impl<S: Append + 'static> Coordinator<S> {
                 // `tick()` on `Interval` is cancel-safe:
                 // https://docs.rs/tokio/1.19.2/tokio/time/struct.Interval.html#cancel-safety
                 _ = advance_timelines_interval.tick() => Message::AdvanceTimelines,
+                _ = storage_usage_update_interval.tick() => Message::StorageUsage,
```

Comment:
oh! I'm not as familiar with this code, but will this block other coord messages from being processed while we're calculating sizes? if so, I think we'll want to move the storage usage work into a separate task

Reply:
@mjibson - Any thoughts on Paul's question? My assumption was that this was non-blocking but I have not proved that. My understanding is that this fires off a message that is picked up in the message processing loop and processed async there.

Reply:
The question is if the call to `shard_sizes()` can take a long time or not at https://github.com/MaterializeInc/materialize/blob/148822a591e00847276a3333ec58e7a6b02d74bb/src/adapter/src/coord/message_handler.rs#L79

If that takes more than 1ms to return, then we should add a new `Message::StorageUsageSizes(..)` or something and do the shard size fetch work in a separate tokio task (started by `storage_usage_update`).

Reply:
That makes sense. Is this likely to cause immediate performance issues, or is it a change we could bundle in with some future work? There are some follow-up tasks needed for this table before GA already.

Reply:
How long does shard_sizes take to return? 1s? 30s? Whatever that time is is how long user queries will be blocked for. If it's, say, less than 5s then it's ok if nothing gets done for a week because this only runs once per hour. If it takes longer than that then we should bump the time to 24h immediately and then also do the background task work. I just tried to type that up myself but there are ownership issues with the storage usage client that have to be fixed first.

Reply:
> How long does shard_sizes take to return? 1s? 30s? Whatever that time is is how long user queries will be blocked for. If it's, say, less than 5s then it's ok if nothing gets done for a week because this only runs once per hour. If it takes longer than that then we should bump the time to 24h immediately and then also do the background task work.

Probably not more than a few seconds, but it really depends how many shards there are. The method lists every single object in S3, which could be many S3 API calls if there are tens or hundreds of thousands of blobs.

>  I just tried to type that up myself but there are ownership issues with the storage usage client that have to be fixed first.

The `StorageUsageClient` could trivially implement `Clone`. Should be easy enough to just clone up a copy for the task.

Reply:
I've carded up the work to look into this, but the cloud team can't pick up the work right away. Given these concerns, would it make sense to temporarily turn off or comment out the collection interval timer until the user query blocking question can be addressed?

Reply:
Resolved in #14205